pact/pact.py

```diff
@@ -0,0 +1,244 @@
+    """
+
+    HEADERS = {'X-Pact-Mock-Service': 'true'}
+
+    def __init__(self, consumer, provider, host_name='localhost', port=1234):
+        """
+        Constructor for Pact.
+
+        :param consumer: The consumer for this contract.
+        :type consumer: pact.Consumer
+        :param provider: The provider for this contract.
+        :type provider: pact.Provider
+        :param host_name: The host name where the mock service is running.
+        :type host_name: str
+        :param port: The port number where the mock service is running.
+        :type port: int
+        """
+        self.BASE_URI = 'http://{host_name}:{port}'.format(
+            host_name=host_name, port=port)
+
+        self.consumer = consumer
+        self.provider = provider
+        self._description = None
+        self._provider_state = None
+        self._request = None
+        self._response = None
+        self._scenario = None
+
+    def given(self, provider_state):
+        """
+        Define the provider state for this pact.
+
+        When the provider verifies this contract, they will use this field to
+        setup pre-defined data that will satisfy the response expectations.
+
+        :param provider_state: The short sentence that is unique to describe
+            the provider state for this contract.
+        :type provider_state: basestring
+        :rtype: Pact
+        """
+        self._provider_state = provider_state
+        return self
+
+    def upon_receiving(self, scenario):
+        """
+        Define the name of this contract.
+
+        :param scenario: A unique name for this contract.
+        :type scenario: basestring
+        :rtype: Pact
+        """
+        self._description = scenario
+        return self
+
+    def with_request(self, method, path, body=None, headers=None, query=None):
+        """
+        Define the request the request that the client is expected to perform.
+
+        :param method: The HTTP method.
+        :type method: str
+        :param path: The path portion of the URI the client will access.
+        :type path: str
+        :param body: The request body, can be a string or an object that will
+            serialize to JSON, like list or dict, defaults to None.
+        :type body: list, dict or None
+        :param headers: The headers the client is expected to include on with
+            this request. Defaults to None.
+        :type headers: dict or None
+        :param query: The query options the client is expected to send. Can be
+            a dict of keys and values, or a URL encoded string.
+            Defaults to None.
+        :type query: dict, basestring, or None
+        :rtype: Pact
+        """
+        self._request = Request(
+            method, path, body=body, headers=headers, query=query).json()
+
+        return self
+
+    def will_respond_with(
+            self, status, headers=None, body=None):
```

Comment:
Is this newline here part of the style guide? Seems that full-length 106 would be shorter than e.g. 81.